staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go

```diff
@@ -820,8 +820,8 @@ type CDISpec struct {
 	// +kubebuilder:validation:Enum=RemoveWorkloads;BlockUninstallIfWorkloadsExist
 	// CDIUninstallStrategy defines the state to leave CDI on uninstall
 	UninstallStrategy *CDIUninstallStrategy `json:"uninstallStrategy,omitempty"`
-	// Rules on which nodes CDI infrastructure pods will be scheduled
-	Infra sdkapi.NodePlacement `json:"infra,omitempty"`
+	// Rules that should apply to CDI infrastructure
+	Infra *ComponentConfig `json:"infra,omitempty"`
```

Comment:
So we've had a somewhat similar PR before - https://github.com/kubevirt/containerized-data-importer/pull/2563
The main issue was breaking API compatibility - https://github.com/kubevirt/containerized-data-importer/pull/2563#discussion_r1092025583

Reply:
https://github.com/kubevirt/containerized-data-importer/pull/2933#discussion_r1366348583
@akalenyu Hi, think you for your review, in this review awels suggest just adding the replicas to the CDISpec struct. In this method API compatibility will not be  broken. Do you think this is okay?

Reply:
I think as is it will still break between upgrades
old version has `spec.Infra.Affinity` while new version will have `spec.Infra.NodePlacement.Affinity`